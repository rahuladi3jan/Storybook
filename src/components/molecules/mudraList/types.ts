import { ColorVariant, Sizes, Weight } from "../../atoms/mudraTypography/types";

export enum LIST_TYPE {
  ORDERED = "ordered",
  UNORDERED = "unordered",
}
export type listType = "ordered" | "unordered";
export interface IMudraListProps {
  data: string[];
  listType?: listType;
  listStyleType?: string;
  textColor?: ColorVariant;
  size?: Sizes;
  weight?: Weight;
  listStyles?: React.CSSProperties;
}
