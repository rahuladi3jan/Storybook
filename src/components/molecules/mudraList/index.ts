import MudraList from "./mudraList";
export { MudraList };
