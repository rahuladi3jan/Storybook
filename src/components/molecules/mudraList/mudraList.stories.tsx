import React from "react";
import { Meta } from "@storybook/react";

import MudraList from "./mudraList";

export default {
  component: MudraList,
  title: "Molecules/Mudra List",
  parameters: {
    componentSubtitle: "Storybook Mudra List",
    docs: {
      description: {
        component:
          "MudraList enables the incorporation of every conceivable list variant from your Figma design using props exclusively. Each prop is equipped with value suggestions, streamlining the customization process.",
      },
    },
  },
} as Meta;

export const AOrderedList = () => {
  const data = [
    "Ordered List Item 1",
    "Ordered List Item 2",
    "Ordered List Item 3",
  ];

  return <MudraList data={data} />;
};

export const BBasicOrderedList = () => {
  return (
    <MudraList
      listType="ordered"
      data={[
        "Ordered List Item 1",
        "Ordered List Item 2",
        "Ordered List Item 3",
      ]}
      listStyleType="lower-alpha"
      textColor="neutral100"
      size={16}
      weight="regular"
    />
  );
};

export const CBasicUnorderedList = () => {
  const data = [
    "Unordered List Item 1",
    "Unordered List Item 2",
    "Unordered List Item 3",
  ];

  return (
    <MudraList
      listType="unordered"
      data={data}
      listStyleType="circle"
      textColor="neutral100"
      size={16}
      weight="regular"
    />
  );
};

export const DStyledUnorderedList = () => {
  const data = [
    "Styled Unordered List Item 1",
    "Styled Unordered List Item 2",
    "Styled Unordered List Item 3",
  ];

  return (
    <MudraList
      listType="unordered"
      data={data}
      listStyleType="square"
      textColor="storybookOrange100"
      size={18}
      weight="extraBold"
      listStyles={{ color: "blue", margin: 10 }}
    />
  );
};

//defining order of stories based on alphabetical order
AOrderedList.storyName = "Mudra List";
BBasicOrderedList.storyName = "Basic Ordered List";
CBasicUnorderedList.storyName = "Basic Unordered List";
DStyledUnorderedList.storyName = "Styled Unordered List";
