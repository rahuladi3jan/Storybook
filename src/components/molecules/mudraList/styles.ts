import styled, { css } from "styled-components";
import { IMudraListProps } from "./types";

export const UnorderedList = styled.ul<IMudraListProps>`
  ${({ listStyleType }) => css`
    list-style-type: ${listStyleType};
  `}
`;

export const OrderedList = styled.ol<IMudraListProps>`
  ${({ listStyleType }) => css`
    list-style-type: ${listStyleType};
  `}
`;
