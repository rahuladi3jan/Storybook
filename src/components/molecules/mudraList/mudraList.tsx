import React from "react";
import { MudraTypography } from "../../atoms/mudraTypography";
import { UnorderedList, OrderedList } from "./styles";
import { IMudraListProps, LIST_TYPE } from "./types";

function MudraList(props: IMudraListProps) {
  const { data, listType, listStyleType, textColor, size, weight, listStyles } =
    props;
  return (
    <div>
      {listType === LIST_TYPE.ORDERED ? (
        <>
          <OrderedList listStyleType={listStyleType}>
            {data?.map((dataItem, index) => {
              return (
                <li key={"mudra-ordered-list-" + index} style={listStyles}>
                  <MudraTypography
                    textColor={textColor}
                    size={size}
                    weight={weight}
                  >
                    {dataItem}
                  </MudraTypography>
                </li>
              );
            })}
          </OrderedList>
        </>
      ) : (
        <>
          <UnorderedList listStyleType={listStyleType}>
            {data?.map((dataItem, index) => {
              return (
                <li key={"mudra-unordered-list-" + index} style={listStyles}>
                  <MudraTypography
                    textColor={textColor}
                    size={size}
                    weight={weight}
                  >
                    {dataItem}
                  </MudraTypography>
                </li>
              );
            })}
          </UnorderedList>
        </>
      )}
    </div>
  );
}

MudraList.defaultProps = {
  listType: "unordered",
  listStyleType: "disc",
  textColor: "neutral100",
  size: 16,
  weight: "regular",
};

export default MudraList;
