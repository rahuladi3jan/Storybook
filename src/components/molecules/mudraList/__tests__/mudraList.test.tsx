import renderer from "../../../../utils/testRenderer";

import * as tests from "../mudraList.stories";

renderer(tests);
