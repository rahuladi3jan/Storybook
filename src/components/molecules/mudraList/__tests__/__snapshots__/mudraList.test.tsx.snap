// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Mudra List AOrderedList 1`] = `
<div>
  <ul
    className="sc-bwzfXH gtusOf"
  >
    <li>
      <span
        className="sc-bdVaJa kPRpeD"
        size={16}
      >
        Ordered List Item 1
      </span>
    </li>
    <li>
      <span
        className="sc-bdVaJa kPRpeD"
        size={16}
      >
        Ordered List Item 2
      </span>
    </li>
    <li>
      <span
        className="sc-bdVaJa kPRpeD"
        size={16}
      >
        Ordered List Item 3
      </span>
    </li>
  </ul>
</div>
`;

exports[`Molecules/Mudra List BBasicOrderedList 1`] = `
<div>
  <ol
    className="sc-htpNat jjzYvY"
  >
    <li>
      <span
        className="sc-bdVaJa kPRpeD"
        size={16}
      >
        Ordered List Item 1
      </span>
    </li>
    <li>
      <span
        className="sc-bdVaJa kPRpeD"
        size={16}
      >
        Ordered List Item 2
      </span>
    </li>
    <li>
      <span
        className="sc-bdVaJa kPRpeD"
        size={16}
      >
        Ordered List Item 3
      </span>
    </li>
  </ol>
</div>
`;

exports[`Molecules/Mudra List CBasicUnorderedList 1`] = `
<div>
  <ul
    className="sc-bwzfXH lgBAof"
  >
    <li>
      <span
        className="sc-bdVaJa kPRpeD"
        size={16}
      >
        Unordered List Item 1
      </span>
    </li>
    <li>
      <span
        className="sc-bdVaJa kPRpeD"
        size={16}
      >
        Unordered List Item 2
      </span>
    </li>
    <li>
      <span
        className="sc-bdVaJa kPRpeD"
        size={16}
      >
        Unordered List Item 3
      </span>
    </li>
  </ul>
</div>
`;

exports[`Molecules/Mudra List DStyledUnorderedList 1`] = `
<div>
  <ul
    className="sc-bwzfXH jpLtKc"
  >
    <li
      style={
        Object {
          "color": "blue",
          "margin": 10,
        }
      }
    >
      <span
        className="sc-bdVaJa hsKQWw"
        size={18}
      >
        Styled Unordered List Item 1
      </span>
    </li>
    <li
      style={
        Object {
          "color": "blue",
          "margin": 10,
        }
      }
    >
      <span
        className="sc-bdVaJa hsKQWw"
        size={18}
      >
        Styled Unordered List Item 2
      </span>
    </li>
    <li
      style={
        Object {
          "color": "blue",
          "margin": 10,
        }
      }
    >
      <span
        className="sc-bdVaJa hsKQWw"
        size={18}
      >
        Styled Unordered List Item 3
      </span>
    </li>
  </ul>
</div>
`;
